src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -76,54 +83,123 @@ public Shooter(OperatorInterface operatorinterface) {
         shooter.setNeutralMode(NeutralMode.Coast);
         shooter.configSupplyCurrentLimit(shooterLimit);
 
+        ml.createfile("shooterRPMS");
+
+        try {
+            shooterRpms = Double.parseDouble(ml.readFile());
+        }
+
+        catch (NullPointerException e) {
+            e.printStackTrace();
+        }
+
+        catch (NumberFormatException e) {
+            e.printStackTrace();
+        }
+
     }
 
     public void runShooter() {
         // System.out.println(shooter.getSelectedSensorVelocity()); PRINTS FOR VELOCITY
         // CONTROL
 
         // Control for FlyWheel
+
         if (oi.runFlyWheelButtonManual()) {
-            startShooter(shooterRpms);
-        } else {
+            shooterState = ShooterState.MANUAL;
+            startShooter();
+        }
+
+        else if (oi.autoShootButton()) {
+            if (checkVision()) {
+                shooterState = ShooterState.AUTOMATIC;
+            }
+
+            else {
+                shooterState = ShooterState.MANUAL;
```

Comment:
You need to maintain the last reading not switch to a separate level if there is a miss on the vision

Reply:
resolved